### PR TITLE
xds: Avoid logging and throwing errors

### PR DESCRIPTION
**Pull request by ejona86**
The FINE logging was just repeating the exceptions. But really, it is
trivial to avoid exceptions in this case and that is beneficial because
it will avoid an expensive error handling path in something that is
trivial to trigger remotely.

The WARNING may be a bit much if connections don't match the filter
chains often in production, but it seems most likely a misconfiguration
and not something that would be seen often.

CC @sanjaypujare

### CODE REVIEW

**Comment by sanjaypujare** [comment]
Wondering why have the message "This should not be possible as ClientXdsClient validated the chains for uniqueness." in the exception instead of just a comment in the code.

> The WARNING may be a bit much if connections don't match the filter chains often in production, but it seems most likely a misconfiguration and not something that would be seen often.

Can happen with DoS or such attacks?

File: xds/src/main/java/io/grpc/xds/FilterChainMatchingProtocolNegotiators.java [hunk 1]
```diff
@@ -149,9 +150,8 @@ SelectedConfig select(InetSocketAddress localAddr, InetSocketAddress remoteAddr)
         filterChains = filterOnSourcePort(filterChains, remoteAddr.getPort());
 
         if (filterChains.size() > 1) {
-          log.log(Level.FINE, "Found more than one matching filter chains: {0}", filterChains);
-          throw new IllegalStateException("Found more than one matching filter chains.");
-          // TODO(chengyuanzhang): should we just return any matched one?
+          throw new IllegalStateException("Found more than one matching filter chains. This should "
+              + "not be possible as ClientXdsClient validated the chains for uniqueness.");
```
Comment (YifeiZhuang) [inline xds/src/main/java/io/grpc/xds/FilterChainMatchingProtocolNegotiators.java:154]: > `ClientXdsClient`

do you mean xds in general, or specifically this network filter chain match?

**Review by YifeiZhuang — APPROVED** [review]
Thanks for the fix.
Sry I closed https://github.com/grpc/grpc-java/pull/8333 too early, I thought you've done review.

**Comment by ejona86** [comment]
> Wondering why have the message "This should not be possible as ClientXdsClient validated the chains for uniqueness." in the exception instead of just a comment in the code.

Basically, it is useful to say "This is a bug" so that users know it isn't their fault and they should file a bug. I could say "this is a bug" in the exception and put a comment about more details. I don't really care much one way or another.

> Can happen with DoS or such attacks?

Yes, it could fill up logs. A logging rate-limit would address that if it was a larger concern. Although the attacker has to get lucky and not match a filter chain. With IP spoofing that could be a bit easier, but by-and-large it seems unlikely there's a configuration that exposes a large surface here.

I'll note the previous code that used an exception was much more expensive, as it had to create the exception and would end up printing out the entire exception (indirectly, via `ctx.fireExceptionCaught()`). So this is still strictly-better. 

> Sry I closed #8333 too early, I thought you've done review.

You're fine. I did hope to review it again, but I had been tied up with retries and there were so many comments back-and-forth that a request for me to re-review would have probably been lost as well. It did look pretty good, and just the log+propagate exception was the only thing that looked off. And since it was a FINE log, it didn't matter much.

I considered even not sending this, but the FINE log was doing string concatenation instead of `{0}`-style and one thing leads to another...

File: xds/src/main/java/io/grpc/xds/FilterChainMatchingProtocolNegotiators.java (hunk 1 above)
Comment (ejona86) [reply xds/src/main/java/io/grpc/xds/FilterChainMatchingProtocolNegotiators.java:154]: I mean specifically `ClientXdsClient.checkForUniqueness()`, but that's a private method.

**Pull request merged**